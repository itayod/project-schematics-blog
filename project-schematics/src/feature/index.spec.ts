import { SchematicTestRunner, UnitTestTree } from '@angular-devkit/schematics/testing';
import * as path from 'path';
import {
  getTestProjectPath,
  createWorkspace,
  defaultWorkspaceOptions,
  defaultAppOptions
} from '../utils/create-workspace';

const collectionPath = path.join(__dirname, '../collection.json');

describe('Feature', () => {
  const schematicRunner = new SchematicTestRunner('schematics', collectionPath);

  let appTree: UnitTestTree;

  beforeEach(() => {
    appTree = createWorkspace(schematicRunner, appTree);
  });

  it('should create a proper module', () => {
    const options = {
      name: 'foo',
      project: 'baz'
    };

    const specifiedProjectPath = getTestProjectPath(defaultWorkspaceOptions, {
      ...defaultAppOptions,
      name: 'baz'
    });

    const tree = schematicRunner.runSchematic('feature', options, appTree);
    const content = tree.readContent(`${specifiedProjectPath}/src/lib/foo/foo.module.ts`);

    expect(content).toMatchSnapshot();
  });
});
