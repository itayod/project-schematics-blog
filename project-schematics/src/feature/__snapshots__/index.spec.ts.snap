// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature should create a proper module 1`] = `
"import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { FooRoutingModule } from './foo-routing.module';
import { FooComponent } from './foo.component';
import { StoreModule } from '@ngrx/store';
import * as fromFoo from './foo.reducer';
import { EffectsModule } from '@ngrx/effects';
import { FooEffects } from './foo.effects';

@NgModule({
  declarations: [FooComponent],
  imports: [
    CommonModule,
    FooRoutingModule,
    StoreModule.forFeature('foo', fromFoo.reducer),
    EffectsModule.forFeature([FooEffects])
  ]
})
export class FooModule { }
"
`;
